parquet/src/main/java/org/apache/iceberg/parquet/ParquetWriter.java

```diff
@@ -197,9 +198,11 @@ private void startRowGroup() {
 
     PageWriteStore pageStore = pageStoreCtorParquet.newInstance(
         compressor, parquetSchema, props.getAllocator(), this.columnIndexTruncateLength);
+    Preconditions.checkState(pageStore instanceof BloomFilterWriteStore,
+        "pageStore must be an instance of BloomFilterWriteStore");
```

Comment:
Do we need this check? It throws a `ClassCastException` at line 205 anyway.

Reply:
The code would be more readable when we add the check here.

Reply:
I'm a little confused on how the type might not match here? We get the class due to reflection so do we have a chance here of getting the wrong class?

Seems like we may benefit from moving this into the DynamicConstructor code at the top of the file? I may misunderstand this but it seems like we should try to do this in the reflection itself rather than after we instantiate?

Reply:
Yes, the check is redundant here. I'll remove it.

> Seems like we may benefit from moving this into the DynamicConstructor code at the top of the file?

We are checking the constructed instance against some interface type. Maybe it will help if we add constraint to `DynamicConstructor `.